pulp_smash/pulp3/pytest_plugin/__init__.py

```diff
@@ -29,7 +38,67 @@
 PULP_SERVICES = ("pulpcore-content", "pulpcore-api", "pulpcore-worker@1", "pulpcore-worker@2")
 
 
+## Pulp No Leftover Objs Feature
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup("pulp-smash")
+    group.addoption(
+        "--pulp-no-leftover-objs",
+        action="store_true",
+        dest="pulp_no_leftover_objs",
+        help="Enable this to have Pulp plugins check for objects leftover by tests.",
+    )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_check_for_leftover_pulp_objects(config):
+    pulpcore_client = gen_pulpcore_client()
+    tasks_api_client = TasksApi(pulpcore_client)
+
+    for task in tasks_api_client.list().results:
+        if task.state in ["running", "waiting"]:
+            raise Exception(f"This test left over a task in the running or waiting state.")
+
+    types_to_check = [
+        ContentguardsApi(pulpcore_client),
+        DistributionsApi(pulpcore_client),
+        PublicationsApi(pulpcore_client),
+        RepositoriesApi(pulpcore_client),
+    ]
+    for type_to_check in types_to_check:
+        if type_to_check.list().count > 0:
+            raise Exception(f"This test left over a {type_to_check}.")
```

Comment:
is `repr(type_to_check)` readable? Also maybe call it `model_to_check`.

Reply:
I switched this to model_to_check in my next push.

Reply:
I think it's pretty readable it produces a statement like: `This test left over a <class 'pulpcore.client.pulpcore.api.distributions_api.DistributionsApi'>.`